src/adams_methods.jl

```diff
@@ -0,0 +1,655 @@
+    h = initstep # current trial stepsize for a give step
+    prev_h = h # the stepsize of the last successful step
+    next_h = 0.0 # stepsize to be used in next step
+
+    if h == 0.
+        # initial guess at a step size
+        # h, tdir, dy0 = hinit(F, y0, tstart, tfinal, 3, reltol, abstol)
+        h, tdir, dy0 = hinit(F, y0, tstart, tfinal, 1, reltol, abstol)
+
+    else
+        tdir = sign(tfinal - tstart)
+    end
+
+    n = 1     #`n` counts the time steps. We start at n=1, the first step
+    steporder=1 # `steporder` is the order of the AB method at the current step
+                # Note: that due to Julia starting indexing at 1, whereas Hairer
+                # indices his variable for order, k, from 0, we have
+                # k = steporder + 1 in what follows
+
+    # initialization of arrays used in estimation y_np1 given dy[n],y[n], h and
+    # list order timestep are  initialized to a maxorder x maxorder dimmension
+    c,g,b,ϕstar, ϕ = intialize_coef_arrays(max_order+3,Tdy)
+
+    timeout_VS_const = 0 # How many steps before a variable step size can be taken
+    timeout_VS = timeout_VS_const # `timeout_VS` is a counter for this in the loop
+
+    timeout_VO_const = 2 # How many steps before a variable step order can be taken
+    timeout_VO = timeout_VO_const # `timeout_VO` is a counter for this in the loop
+
+    consec_const_steps = 0 # counts how many consecutives steps have been taken
+                           # with same step size
+
+    # The four local error (LE) or truncation error necessary for variable
+    # order calculations are listed below, which correspond to infinity norms of:
+    #      LE(n+1)_{k-1}, LE(n+1)_{k}, E(n+1)_{k+1}, and LE(n+1)_{k+2}
+    # from Harier pgs 422-424
+    err_km1, err_k, err_kp1, err_kp2 = 0.0,0.0,0.0,0.0
+
+    #-##########################################################################
+    # Variable Step Variable Order (VSVO) Step Integration Loop
+    #-##########################################################################
+    while  tfinal*tdir > t[n]*tdir
+        #--#####################################################################
+        # Attempt a step with current trial stepsize `h`
+        #--#####################################################################
+        successful_step = false
+        while !successful_step
+            #---################################################################
+            # Make an approriate time step with current trial stepsize `h`
+            #---################################################################
+            #Keep step size constant if `h` is close to `prev_h`
+            if abs(h/prev_h)<2 && abs(h/prev_h)>=1
+                consec_const_steps += 1
+                h = prev_h
+            else
+                consec_const_steps = 0
+            end
+
+            #Don't step past `tfinal`
+            if tdir*(t[n] + h) >= tdir*(tfinal)
+                h = tfinal-t[n]
```

Comment:
Shouldn't `consec_const_steps` be updated here too?  Maybe it would be better to do this check when calculating the new `h`?